### PR TITLE
[LLM Runtime] Fix the accuracy issue with group_size=-1

### PR DESCRIPTION
**Pull request by luoyu-intel**
## Type of Change

update jblas' bug fix commits

### CODE REVIEW

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/229/

**Comment by VincyZhang** [comment]
windows binary build: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-release-build-win/299/

**Review by airMeng — APPROVED** [review]

File: intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx512f.h [hunk 1]
```diff
@@ -959,7 +973,7 @@ inline void f32_f4_quantize_1x16(const float* srcptr, int8_t* dstptr, int ld_src
     if constexpr (F4_T == JBLAS_DTYPE::F4_BNB) sub_v = _mm512_set1_ps(F4_BNB_quant_sub_helper[i]);
     if constexpr (F4_T == JBLAS_DTYPE::F4_E2M1) sub_v = _mm512_set1_ps(F4_E2M1_quant_sub_helper[i]);
     zmm1 = _mm512_sub_ps(zmm0, sub_v);
-    mask0 = _mm512_cmple_ps_mask(zmm1, zmm_v0);
+    mask0 = _mm512_cmp_ps_mask(zmm1, zmm_v0, 2);
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx512f.h:976]: what's the difference?

File: intel_extension_for_transformers/llm/library/jblas/jblas/kernel_ref.h [hunk 2]
```diff
@@ -230,25 +230,47 @@ inline void convert_s4_s8_8(int8_t* dstptr, int8_t* srcptr) {
...
+}
+
+template <>
+inline void convert_s4_s8_8<JBLAS_DTYPE::F4_NF4>(int8_t* dstptr, int8_t* srcptr) {
+  convert_s4_s8_8_lowbits(dstptr, srcptr);
+}
+
+template <>
+inline void convert_s4_s8_8<JBLAS_DTYPE::F4_E2M1>(int8_t* dstptr, int8_t* srcptr) {
+  convert_s4_s8_8_lowbits(dstptr, srcptr);
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/library/jblas/jblas/kernel_ref.h:273]: exactly same?

File: intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx512f.h (hunk 1 above)
Comment (zhewang1-intc) [reply intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx512f.h:976]: we found `_mm512_cmple_ps_mask` can't pass compile on Red Hat default gcc-toolset (gcc version 8.3/9.3) with Debug flag.. 
the strange thing is that the compile error will disapper if with Relase flag.

File: intel_extension_for_transformers/llm/library/jblas/jblas/kernel_ref.h (hunk 2 above)
Comment (luoyu-intel) [reply intel_extension_for_transformers/llm/library/jblas/jblas/kernel_ref.h:273]: yes, but look up different tables.

**Comment by DDEle** [comment]
It is confirmed that this PR benefits MLPerf accuracy:
- reference: "ROUGE1", 42.9865 * 0.99, "ROUGE2", 20.1235 * 0.99, "ROUGEL", 29.9881 * 0.99, "GEN_LEN", 4016878*0.9
- q4_j1_bf16_pc-gcc12 (#322):
  {'rouge1': 43.3352, 'rouge2': 20.1831, 'rougeL': 29.8832, 'rougeLsum': 40.4381, 'gen_len': 4414947, 'gen_num': 13368}
- **q4_j1_bf16_pc-no_mha-gcc13 (PR1057, 369c597) (all)**
  **{'rouge1': 43.2917, 'rouge2': 20.1802, 'rougeL': 29.9402, 'rougeLsum': 40.4317, 'gen_len': 4413826, 'gen_num': 13368}**

**Pull request merged**